src/components/layout/header/Search.tsx

```diff
@@ -142,13 +143,25 @@ export function Search() {
   const [OverseerrResults, setOverseerrResults] = useState<any[]>([]);
   const [opened, setOpened] = useState(false);
 
-  useEffect(() => {
-    if (debounced !== '' && selectedSearchEngine.value === 'overseerr' && searchQuery.length > 3) {
-      axios.get(`/api/modules/overseerr?query=${searchQuery}`).then((res) => {
-        setOverseerrResults(res.data.results ?? []);
-      });
+  const { data, isLoading, error } = useQuery(
+    ['overseerr', debounced],
+    async () => {
+      if (debounced !== '' && selectedSearchEngine.value === 'overseerr' && debounced.length > 3) {
+        const res = await axios.get(`/api/modules/overseerr?query=${debounced}`);
+        return res.data.results ?? [];
+      }
+      return [];
+    },
+    {
+      refetchOnWindowFocus: false,
+      refetchOnMount: false,
+      refetchInterval: false,
     }
-  }, [debounced]);
+  );
+
+  useEffect(() => {
+    setOverseerrResults(data ?? []);
+  }, [data]);
 
   const isModuleEnabled = config?.settings.customization.layout.enabledSearchbar;
```

Comment:
Rename variable

Reply:
It's the one we used before too and it's pretty descriptive

Reply:
No. I would call it ``isIntegrationEnabled``, as there are no modules anymore

Reply:
Oh you meant that line I thought the OverseerrResult one